File: MediumRecyclerViewKullanimi/app/src/main/java/com/example/mediumrecyclerviewkullanimi/ui/adapter/CountryAdapter.java
package com.example.mediumrecyclerviewkullanimi.ui.adapter;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.example.mediumrecyclerviewkullanimi.data.entity.Country;
import com.example.mediumrecyclerviewkullanimi.databinding.CardDesignBinding;

import java.util.List;

public class CountryAdapter extends RecyclerView.Adapter<CountryAdapter.CountryCardDesignHolder> {

    private List<Country> countryListesi;
    private Context mContext;

    public CountryAdapter(List<Country> countryListesi, Context mContext) {
        this.countryListesi = countryListesi;
        this.mContext = mContext;
    }
    public class CountryCardDesignHolder extends RecyclerView.ViewHolder{
        private CardDesignBinding binding;

        public CountryCardDesignHolder(CardDesignBinding binding){
            super(binding.getRoot());
            this.binding = binding;
        }
    }

    @NonNull
    @Override
    public CountryCardDesignHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        CardDesignBinding binding = CardDesignBinding.inflate(LayoutInflater.from(mContext),parent,false);
        return new CountryCardDesignHolder(binding);
    }

    @Override
    public void onBindViewHolder(@NonNull CountryCardDesignHolder holder, int position) {
        Country country = countryListesi.get(position);
        CardDesignBinding binding = holder.binding;

        binding.ivCountry.setImageResource(mContext.getResources().getIdentifier(country.getCountryImage(),"drawable",mContext.getPackageName()));
        binding.tvCountryName.setText(country.getCountryName());

    }

    @Override
    public int getItemCount() {
        return countryListesi.size();
    }

}

File: MediumRecyclerViewKullanimiKotlin/app/src/main/java/com/example/mediumrecyclerviewkullanimikotlin/ui/adapter/CountryAdapter.kt
package com.example.mediumrecyclerviewkullanimikotlin.ui.adapter

import android.content.Context
import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import com.example.mediumrecyclerviewkullanimikotlin.data.entity.Country
import com.example.mediumrecyclerviewkullanimikotlin.databinding.CardDesignBinding

class CountryAdapter(private val countryListesi: List<Country>, private val mContext: Context) :
    RecyclerView.Adapter<CountryAdapter.CountryCardDesignHolder>() {

    inner class CountryCardDesignHolder(val binding: CardDesignBinding) :
        RecyclerView.ViewHolder(binding.root)

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): CountryCardDesignHolder {
        val binding = CardDesignBinding.inflate(LayoutInflater.from(mContext), parent, false)
        return CountryCardDesignHolder(binding)
    }

    override fun onBindViewHolder(holder: CountryCardDesignHolder, position: Int) {
        val country = countryListesi[position]
        val binding = holder.binding

        binding.ivCountry.setImageResource(
            mContext.resources.getIdentifier(
                country.countryImage,
                "drawable",
                mContext.packageName
            )
        )
        binding.tvCountryName.text = country.countryName
    }

    override fun getItemCount(): Int {
        return countryListesi.size
    }
}
